### PR TITLE
Feature/mc 12506 updating api documentation for kubernetes storage entities

### PR DESCRIPTION
**Pull request by danielvanacker**
#### Changes made

-Added documentation in Kubernetes and Kubernetes_extension for updating storage classes, persistent volumes, and persistent volume claims

#### Related PRs
- [#188 ](https://github.com/cloudops/cloudmc-kubernetes-sdk/pull/188)

### CODE REVIEW

File: source/includes/kubernetes/_k8_persistentvolumeclaims.md [hunk 1]
```diff
@@ -223,6 +223,77 @@ Return value:
...
+<code>PUT /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/persistentvolumeclaims/:id</code>
+
+Replace a persistent volume claim in a given [environment](#administration-environments).
+
+Required Attributes                 | &nbsp;
+----------------------------------- | ------------------------------------------------------------
+`apiVersion` <br/>_string_          | The api version (versioned schema) of the pod.
+`metadata` <br/>_object_            | The metadata of the pod.
+`metadata.name` <br/>_string_       | The name of the pod.
+`spec`<br/>_object_                 | The specification used to replace and run the pod.
```
Comment (m-orfanos) [inline source/includes/kubernetes/_k8_persistentvolumeclaims.md]: `pod` -> `persistent volume claim`

File: source/includes/kubernetes/_k8_persistentvolumeclaims.md [hunk 2]
```diff
@@ -223,6 +223,77 @@ Return value:
...
+| `spec.volumemode` <br/>_string_        | What type of volume is required by the claim. Value of Filesystem is implied when not included in claim spec.|
+| `spec.dataSource` <br/>_string_        | This field can be used to specify either: an existing VolumeSnapshot object, an existing PVC or an existing custom resource/object that implements data population.|
+
+Note that the list is not complete, since it is refering to the [kubernetes api details](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#persistentvolumeclaim-v1-core).
+
+Return value:
+
+| Attributes                 | &nbsp;                                       |
+| -------------------------- | -------------------------------------------- |
+| `taskId` <br/>_string_     | The id corresponding to the replace pod task.|
```
Comment (m-orfanos) [inline source/includes/kubernetes/_k8_persistentvolumeclaims.md]: `pod` -> `persistent volume claim`

File: source/includes/kubernetes/_k8_persistentvolumes.md [hunk 3]
```diff
@@ -238,6 +238,90 @@ Return value:
...
+
+Replace a persistent volume in a given [environment](#administration-environments).
+
+Required Attributes                 | &nbsp;
+----------------------------------- | ------------------------------------------------------------
+`apiVersion` <br/>_string_          | The api version (versioned schema) of the pod.
+`metadata` <br/>_object_            | The metadata of the pod.
+`metadata.name` <br/>_string_       | The name of the pod.
+`spec`<br/>_object_                 | The specification used to replace and run the pod.
+`spec.accessModes`<br/>_string_ | The desired access modes the volume should have.
```
Comment (m-orfanos) [inline source/includes/kubernetes/_k8_persistentvolumes.md]: `pod` -> `persistent volume`

File: source/includes/kubernetes/_k8_persistentvolumes.md [hunk 4]
```diff
@@ -238,6 +238,90 @@ Return value:
...
+| `spec.storageClassName` <br/>_string_        | Name of StorageClass to which this persistent volume belongs. Empty value means that this volume does not belong to any StorageClass.|
+| `spec.volumeMode` <br/>_string_        | Defines if a volume is intended to be used with a formatted filesystem or to remain in raw block state. Value of Filesystem is implied when not included in spec.|
+
+Note that the list is not complete, since it is refering to the [kubernetes api details](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#persistentvolume-v1-core).
+
+Return value:
+
+| Attributes                 | &nbsp;                                       |
+| -------------------------- | -------------------------------------------- |
+| `taskId` <br/>_string_     | The id corresponding to the replace pod task.|
```
Comment (m-orfanos) [inline source/includes/kubernetes/_k8_persistentvolumes.md]: `pod` -> `persistent volume`

File: source/includes/kubernetes/_k8_storageclasses.md [hunk 5]
```diff
@@ -185,6 +185,57 @@ Return value:
...
+```json
+{
+  "taskId": "1542bd45-4732-419b-87b6-4ea6ec695c2b",
+  "taskStatus": "PENDING"
+}
+```
+
+<code>PUT /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/storageclasses/:id</code>
+
+Replace a persistent volume claim in a given [environment](#administration-environments).
```
Comment (m-orfanos) [inline source/includes/kubernetes/_k8_storageclasses.md]: `persistent volume claim` -> `storage class`

File: source/includes/kubernetes/_k8_storageclasses.md [hunk 6]
```diff
@@ -185,6 +185,57 @@ Return value:
...
+
+Required Attributes                 | &nbsp;
+----------------------------------- | ------------------------------------------------------------
+`apiVersion` <br/>_string_          | The api version (versioned schema) of the pod.
+`metadata` <br/>_object_            | The metadata of the pod.
+`metadata.name` <br/>_string_       | The name of the pod.
```
Comment (m-orfanos) [inline source/includes/kubernetes/_k8_storageclasses.md]: `pod` -> `storage class`

File: source/includes/kubernetes/_k8_storageclasses.md [hunk 7]
```diff
@@ -185,6 +185,57 @@ Return value:
...
+| ----------------------------------------- | ----------------------------------------------------------------------- |
+| `reclaimPolicy` <br/>_string_        | Dynamically provisioned PersistentVolumes of this storage class are created with this reclaimPolicy. Defaults to Delete.|
+
+Note that the list is not complete, since it is refering to the [kubernetes api details](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#storageclass-v1-storage-k8s-io).
+
+Return value:
+
+| Attributes                 | &nbsp;                                       |
+| -------------------------- | -------------------------------------------- |
+| `taskId` <br/>_string_     | The id corresponding to the replace pod task.|
```
Comment (m-orfanos) [inline source/includes/kubernetes/_k8_storageclasses.md]: `pod` -> `storage class`

File: source/includes/kubernetes_extension/_k8_persistentvolumeclaims.md [hunk 8]
```diff
@@ -235,6 +235,77 @@ Return value:
...
+| `spec.volumemode` <br/>_string_        | What type of volume is required by the claim. Value of Filesystem is implied when not included in claim spec.|
+| `spec.dataSource` <br/>_string_        | This field can be used to specify either: an existing VolumeSnapshot object, an existing PVC or an existing custom resource/object that implements data population.|
+
+Note that the list is not complete, since it is refering to the [kubernetes api details](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#persistentvolumeclaim-v1-core).
+
+Return value:
+
+| Attributes                 | &nbsp;                                       |
+| -------------------------- | -------------------------------------------- |
+| `taskId` <br/>_string_     | The id corresponding to the replace pod task.|
```
Comment (m-orfanos) [inline source/includes/kubernetes_extension/_k8_persistentvolumeclaims.md]: pod -> persistent volume claim

File: source/includes/kubernetes_extension/_k8_persistentvolumeclaims.md [hunk 9]
```diff
@@ -235,6 +235,77 @@ Return value:
...
+<code>PUT /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/persistentvolumeclaims/:id?cluster_id=:cluster_id</code>
+
+Replace a persistent volume claim in a given [environment](#administration-environments).
+
+Required Attributes                 | &nbsp;
+----------------------------------- | ------------------------------------------------------------
+`apiVersion` <br/>_string_          | The api version (versioned schema) of the pod.
+`metadata` <br/>_object_            | The metadata of the pod.
+`metadata.name` <br/>_string_       | The name of the pod.
+`spec`<br/>_object_                 | The specification used to replace and run the pod.
```
Comment (m-orfanos) [inline source/includes/kubernetes_extension/_k8_persistentvolumeclaims.md]: pod -> persistent volume claim

File: source/includes/kubernetes_extension/_k8_persistentvolumes.md [hunk 10]
```diff
@@ -246,6 +246,89 @@ Return value:
...
+<code>PUT /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/persistentvolumes/:id?cluster_id=a_cluster_id</code>
+
+Replace a persistent volume in a given [environment](#administration-environments).
+
+Required Attributes                 | &nbsp;
+----------------------------------- | ------------------------------------------------------------
+`apiVersion` <br/>_string_          | The api version (versioned schema) of the pod.
+`metadata` <br/>_object_            | The metadata of the pod.
+`metadata.name` <br/>_string_       | The name of the pod.
+`spec`<br/>_object_                 | The specification used to replace and run the pod.
```
Comment (m-orfanos) [inline source/includes/kubernetes_extension/_k8_persistentvolumes.md]: pod -> persistent volume

File: source/includes/kubernetes_extension/_k8_persistentvolumes.md [hunk 11]
```diff
@@ -246,6 +246,89 @@ Return value:
...
+| `spec.volumeMode` <br/>_string_        | Defines if a volume is intended to be used with a formatted filesystem or to remain in raw block state. Value of Filesystem is implied when not included in spec.|
+
+Note that the list is not complete, since it is refering to the [kubernetes api details](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#persistentvolume-v1-core).
+
+
+Return value:
+
+| Attributes                 | &nbsp;                                       |
+| -------------------------- | -------------------------------------------- |
+| `taskId` <br/>_string_     | The id corresponding to the replace pod task.|
```
Comment (m-orfanos) [inline source/includes/kubernetes_extension/_k8_persistentvolumes.md]: pod -> persistent volume

File: source/includes/kubernetes_extension/_k8_storageclasses.md [hunk 12]
```diff
@@ -193,6 +193,57 @@ Return value:
...
+```json
+{
+  "taskId": "1542bd45-4732-419b-87b6-4ea6ec695c2b",
+  "taskStatus": "PENDING"
+}
+```
+
+<code>PUT /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/storageclasses/:id?cluster_id=:cluster_id</code>
+
+Replace a persistent volume claim in a given [environment](#administration-environments).
```
Comment (m-orfanos) [inline source/includes/kubernetes_extension/_k8_storageclasses.md]: persistent volume claim -> storage class

File: source/includes/kubernetes_extension/_k8_storageclasses.md [hunk 13]
```diff
@@ -193,6 +193,57 @@ Return value:
...
+
+Required Attributes                 | &nbsp;
+----------------------------------- | ------------------------------------------------------------
+`apiVersion` <br/>_string_          | The api version (versioned schema) of the pod.
+`metadata` <br/>_object_            | The metadata of the pod.
+`metadata.name` <br/>_string_       | The name of the pod.
```
Comment (m-orfanos) [inline source/includes/kubernetes_extension/_k8_storageclasses.md]: pod -> storage class

File: source/includes/kubernetes_extension/_k8_storageclasses.md [hunk 14]
```diff
@@ -193,6 +193,57 @@ Return value:
...
+| ----------------------------------------- | ----------------------------------------------------------------------- |
+| `reclaimPolicy` <br/>_string_        | Dynamically provisioned PersistentVolumes of this storage class are created with this reclaimPolicy. Defaults to Delete.|
+
+Note that the list is not complete, since it is refering to the [kubernetes api details](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#storageclass-v1-storage-k8s-io).
+
+Return value:
+
+| Attributes                 | &nbsp;                                       |
+| -------------------------- | -------------------------------------------- |
+| `taskId` <br/>_string_     | The id corresponding to the replace pod task.|
```
Comment (m-orfanos) [inline source/includes/kubernetes_extension/_k8_storageclasses.md]: pod -> storage class

**Review by m-orfanos — CHANGES_REQUESTED** [review]
I took a quick look, you have some copy-paste errors :p

**Comment by danielvanacker** [comment]
> I took a quick look, you have some copy-paste errors :p

😁 my bad, just fixed those

File: source/includes/kubernetes_extension/_k8_persistentvolumes.md [hunk 15]
```diff
@@ -246,6 +246,89 @@ Return value:
...
+  "taskId": "1542bd45-4732-419b-87b6-4ea6ec695c2b",
+  "taskStatus": "PENDING"
+}
+```
+
+<code>PUT /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/persistentvolumes/:id?cluster_id=a_cluster_id</code>
+
+Replace a persistent volume in a given [environment](#administration-environments).
+
+Required Attributes                 | &nbsp;
```
Comment (y0geshdev) [inline source/includes/kubernetes_extension/_k8_persistentvolumes.md:305]: I think we use `Attributes` as it implicitly means required attributes. Can we check with other sections of the API docs to make sure that it's consistent across the doc?

File: source/includes/kubernetes_extension/_k8_persistentvolumes.md [hunk 16]
```diff
@@ -246,6 +246,89 @@ Return value:
...
+`metadata` <br/>_object_            | The metadata of the volume.
+`metadata.name` <br/>_string_       | The name of the volume.
+`spec`<br/>_object_                 | The specification used to replace and run the volume.
+`spec.accessModes`<br/>_string_ | The desired access modes the volume should have.
+`spec.capacity.storage`<br/>_string_  | The size of the claim.
+`spec.hostPath.path`<br/>_string_  | The path of the directory on the host.
+
+| Optional Attributes                       | &nbsp;                                                                  |
+| ----------------------------------------- | ----------------------------------------------------------------------- |
+| `metadata.annotations` <br/>_string_        | Annotations is an unstructured key value map stored with a resource that may be set by external tools to store and retrieve arbitrary metadata.|
```
Comment (y0geshdev) [inline source/includes/kubernetes_extension/_k8_persistentvolumes.md:317]:
```suggestion
| `metadata.annotations` <br/>_string_        | Annotations are unstructured key-value map stored with a resource that may be set by external tools to store and retrieve arbitrary metadata.|
```

File: source/includes/kubernetes_extension/_k8_persistentvolumes.md (hunk 15 above)
Comment (danielvanacker) [reply source/includes/kubernetes_extension/_k8_persistentvolumes.md:305]: Its Required attributes in other sections

Comment (y0geshdev) [reply source/includes/kubernetes_extension/_k8_persistentvolumes.md:305]: Thanks for checking that.

**Review by y0geshdev — APPROVED** [review]
Left one minor suggestion but apart from that, it looks good.

File: source/includes/kubernetes_extension/_k8_persistentvolumes.md (hunk 16 above)
Comment (danielvanacker) [reply source/includes/kubernetes_extension/_k8_persistentvolumes.md:317]: I think the first makes for sense to me, Annotations is a key value map that holds multiple keys and values. Whereas annotations.<key> are unstructured key-value pairs.

Comment (y0geshdev) [reply source/includes/kubernetes_extension/_k8_persistentvolumes.md:317]: Yes, that's true. I was thinking more in terms of describing the JSON format and fields in the request payload.
In code, we do store these key-value pairs in Map but in JSON we only have the concept of key-value. WYT?

File: source/includes/kubernetes/_k8_persistentvolumes.md [hunk 17]
```diff
@@ -238,6 +238,90 @@ Return value:
...
+`metadata.name` <br/>_string_       | The name of the volume.
+`spec`<br/>_object_                 | The specification used to replace and run the volume.
+`spec.accessModes`<br/>_string_ | The desired access modes the volume should have.
+`spec.capacity.storage`<br/>_string_  | The size of the claim.
+`spec.hostPath.path`<br/>_string_  | The path of the directory on the host.
+
+| Optional Attributes                       | &nbsp;                                                                  |
+| ----------------------------------------- | ----------------------------------------------------------------------- |
+| `metadata.annotations` <br/>_string_        | Annotations is an unstructured key value map stored with a resource that may be set by external tools to store and retrieve arbitrary metadata.|
+| `spec.claimRef` <br/>_string_        | part of a bi-directional binding between PersistentVolume and PersistentVolumeClaim. Expected to be non-nil when bound.|
```
Comment (m-orfanos) [inline source/includes/kubernetes/_k8_persistentvolumes.md:310]: All the other attributes begin with a capital letter except for this one. 

`part ...` -> `Part ....`

**Review by m-orfanos — APPROVED** [review]
just a small comment, otherwise LGTM!

**Pull request merged**